src/extension/helper_process.c

```diff
@@ -192,40 +194,36 @@ dd_conn *nullable dd_helper_mgr_cur_conn(void)
     return NULL;
 }
 
-static char *_concat_paths(const char *nonnull base, size_t base_len,
-    const char *nonnull file, size_t file_len)
-{
-    bool add_slash = base[base_len - 1] != '/';
-
-    // Since file comes from a literal, assume it has a safe length. To that
-    // add the slash, the base length and the null character.
-    char *ret = safe_pemalloc(base_len, 1, (add_slash ? 2 : 1) + file_len, 1);
-    char *ptr = ret;
-
-    memcpy(ptr, base, base_len);
-    ptr += base_len;
-    if (add_slash) {
-        *ptr++ = '/';
-    }
-
-    memcpy(ptr, file, file_len);
-    ptr[file_len] = '\0';
-
-    return ret;
-}
-
 // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
 bool dd_on_runtime_path_update(zval *nullable old_val, zval *nonnull new_val)
 {
     UNUSED(old_val);
 
+    uid_t uid = getuid();
+    gid_t gid = getgid();
+    char *base = Z_STRVAL_P(new_val);
+    size_t base_len = Z_STRLEN_P(new_val);
+    char *separator = base[base_len - 1] != '/' ? "/" : "";
+
+    size_t sock_name_len =
+        snprintf(NULL, 0, DD_SOCK_PATH_EXTENSION, base, separator, uid, gid) +
+        1;
+    char *sock_name = safe_pemalloc(sock_name_len, sizeof(char), 0, 1);
+    snprintf(sock_name, sock_name_len, DD_SOCK_PATH_EXTENSION, base, separator,
+        uid, gid);
+    sock_name[sock_name_len - 1] = '\0';
     pefree(_mgr.socket_path, 1);
-    _mgr.socket_path = _concat_paths(
-        Z_STRVAL_P(new_val), Z_STRLEN_P(new_val), ZEND_STRL(DD_SOCKET_PATH));
-
+    _mgr.socket_path = sock_name;
+
+    size_t lock_name_len =
+        snprintf(NULL, 0, DD_LOCK_PATH_EXTENSION, base, separator, uid, gid) +
```

Comment:
```suggestion
        snprintf(NULL, 0, DD_LOCK_PATH_FORMAT, base, separator, uid, gid) +
```
And like before I would move the +1 to safe_pemalloc(.., .., 1, 1)